File: integrationtests/workspaces/go/consumer.go
package main

import "fmt"

// ConsumerFunction uses the helper function
func ConsumerFunction() {
	message := HelperFunction()
	fmt.Println(message)
	
	// Use shared struct
	s := &SharedStruct{
		ID:   1,
		Name: "test",
		Value: 42.0,
		Constants: []string{SharedConstant},
	}
	
	// Call methods on the struct
	fmt.Println(s.Method())
	s.Process()
	
	// Use shared interface
	var iface SharedInterface = s
	fmt.Println(iface.GetName())
	
	// Use shared type
	var t SharedType = 100
	fmt.Println(t)
}

File: integrationtests/workspaces/rust/src/consumer.rs
// Consumer module for testing references
use crate::helper::helper_function;
use crate::types::{
    SharedInterface, SharedStruct, SharedType, SHARED_CONSTANT,
};

pub fn consumer_function() {
    // Use the helper function
    let result = helper_function();
    println!("Helper result: {}", result);

    // Use shared struct
    let s = SharedStruct::new("test");
    println!("Struct method: {}", s.method());

    // Use shared interface
    let iface: &dyn SharedInterface = &s;
    println!("Interface method: {}", iface.get_name());

    // Use shared constant
    println!("Constant: {}", SHARED_CONSTANT);

    // Use shared type
    let t: SharedType = String::from("test");
    println!("Type: {}", t);
}
